sdk/appconfiguration/Azure.Data.AppConfiguration/src/FeatureFlagConfigurationSetting.cs

```diff
@@ -0,0 +1,355 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections.Generic;
+using System.Collections.ObjectModel;
+using System.IO;
+using System.Linq;
+using System.Text;
+using System.Text.Json;
+
+namespace Azure.Data.AppConfiguration
+{
+    /// <summary>
+    /// Represents a configuration setting that stores a feature flag value.
+    /// </summary>
+    public class FeatureFlagConfigurationSetting : ConfigurationSetting
+    {
+        internal const string FeatureFlagContentType = "application/vnd.microsoft.appconfig.ff+json;charset=utf-8";
+
+        /// <summary>
+        /// The prefix used for <see cref="FeatureFlagConfigurationSetting"/> setting keys.
+        /// </summary>
+        public static string KeyPrefix { get; } = ".appconfig.featureflag/";
+
+        private string _originalValue;
+        private bool _isValidValue;
+        private string _featureId;
+        private string _description;
+        private string _displayName;
+        private bool _isEnabled;
+        private readonly ObservableCollection<FeatureFlagFilter> _clientFilters;
+
+        internal FeatureFlagConfigurationSetting()
+        {
+            _clientFilters = new ObservableCollection<FeatureFlagFilter>();
+            _clientFilters.CollectionChanged += (_, _) => CheckValidWrite();
+        }
+
+        /// <summary>
+        /// Initializes and instance of the <see cref="FeatureFlagConfigurationSetting"/> using a provided feature id and
+        /// the enabled value.
+        /// </summary>
+        /// <param name="featureId">The identified of the feature flag.</param>
+        /// <param name="isEnabled">The value indicating whether the feature flag is enabled.</param>
+        /// <param name="label">A label used to group this configuration setting with others.</param>
+        public FeatureFlagConfigurationSetting(string featureId, bool isEnabled, string label = null) : base(KeyPrefix + featureId, featureId, label)
```

Comment:
If I understand correctly, this class overwrites the behavior of `GetValue` and `SetValue` of `ConfigurationSetting`, so you could send `null` for the `value`, right? `base(KeyPrefix + featureId, null, label)`

Reply:
Not exactly, because we override the GetValue and SetValue we don't really care what value gets passed into that base type constructor because it would never be used (as our GetValue) implementation would be preferred. So we pass null.

Reply:
yeah so you can pass `null` instead of `featureId` in this line

Reply:
I'll drop the base call altogether.